src/Symfony/Bridge/Twig/Extension/SecurityExtension.php

```diff
@@ -48,13 +53,33 @@ public function isGranted($role, $object = null, $field = null)
         }
     }
 
+    public function getImpersonateExitUrl(string $exitTo = null)
+    {
+        if (null === $this->impersonateUrlGenerator) {
+            return '';
```

Comment:
shouldnt it throw instead?

Reply:
Honestly IDK, `isGranted`, just above, does not throw anything if it's dependency is not set.

My thought was, as the function in `ImpersonateUrlGenerator` return an empty string if you ask the url when you are not impersonating a user, it would be pertinent to return the same thing here.

Reply:
i tend to agree an empty string is the most pragmatic as well as consistent solution :+1: checking state can be done using `ROLE_PREVIOUS_ADMIN` as mentioned.